components/topBar/UserDropdown.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {useState } from "react";
+import { tokens } from "../../constants/color-palette";
+import { PersonRounded, ExpandMore, SettingsRounded, LogoutRounded } from "@mui/icons-material";
+import HorizontalDivider from "../common/HorizontalDivider";
+
+interface UserDropdownProps {
+  theme: string;
+  firstName: string;
+  lastName: string;
+}
+
+
+const UserDropdown = (props: UserDropdownProps) => {
+  const theme = useTheme();
+  const colors = tokens(theme.palette.mode);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const isOpen = Boolean(anchorEl);
+
+  const handleClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <Box data-testid="user-dropdown"> 
+      <Button onClick={handleClick} variant="contained"
+      data-testid="user-dropdown-button"
+       sx={{
+        height: "3.5rem",
+        width: "auto",
+        maxWidth: "20rem",
+        marginRight: "3rem",
+        borderRadius: isOpen ? "1rem 1rem 0 0" : "1rem",
+        color: theme.palette.mode === "light" ? (isOpen ? colors.white[500] : colors.secondary[500]) : colors.white[500],
+        backgroundColor: isOpen ? colors.secondary[500] : theme.palette.mode === "light" ? colors.secondary[100] : alpha(colors.secondary[700], 0.2),
+        boxShadow: "none",
+        "&:hover": {
+          backgroundColor: theme.palette.mode === "light" ? alpha(colors.secondary[200], 0.8) : alpha(colors.secondary[700], 0.8),
+          boxShadow: "none",
+        },
+      }}
+      >
+        <PersonRounded sx={{
+          marginRight: "0.5rem",
+          color: theme.palette.mode === "light" ? (isOpen ? colors.white[500] : colors.secondary[500]) : colors.white[500],
+        }} />
+        <Typography variant={"h4"} sx={{
+          marginRight: "1rem",
+          fontWeight: "600",
+          textTransform: "none",
+          color: theme.palette.mode === "light" ? (isOpen ? colors.white[500] : colors.secondary[500]) : colors.white[500],
+        }}>{props.firstName} {props.lastName}</Typography>
+        <ExpandMore sx={{
+          transform: isOpen ? "rotate(180deg)" : "rotate(0deg)",
+          transition: "transform 0.35s ease-in-out",
+        }} />
+      </Button>
+      <Menu
+        anchorEl={anchorEl}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+        transitionDuration={2}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+        sx={{ 
+          '& .MuiPaper-root': {
+            backgroundColor: colors.secondary[500],
+            borderBottomLeftRadius: "1rem",
+            borderBottomRightRadius: "1rem",  
+            boxShadow: "none",
+            padding: "0.5rem",
+            marginTop: "-0.2rem",
+            backgroundImage: "none",
```

Comment:
Yes because this component from MUI has default boxShadow value (which is not present in Figma) and backgroundImage which was reason why colors of button and dropdown menu didn't match before. Margin is necessary for menu to be connected to button and crete an effect of unfolding